woocommerce/woocommerce-paypal-payments
# Add spinner and disable buttons while switching PUI checkbox

**Issue**: #851

---

### Description

When clicking a checkbox to onboard with creditcards or PUI before onboarding, the button must be clicked three times before the popup window opens.

This happens because when the PUI checkbox is clicked then AJAX request is done, which also changes the button `href` attribute. The PR will add a spinner near the buttons, also it will disable the buttons while AJAX request is running. 

### Steps to Test




1. Set store to germany.
2. Disconnect account.
3. Modify checkbox states.
4. Click “Activate PayPal” or “ Test payments with sandbox”.
5. The spinner should be shown near disabled buttons while AJAX request is in progress.

---

Closes #851 .
